clang/test/SemaCXX/constant-expression-cxx14.cpp

```diff
@@ -265,7 +265,7 @@ namespace const_modify {
 
 namespace null {
   constexpr int test(int *p) {
-    return *p = 123; // expected-note {{assignment to dereferenced null pointer}}
+    return *p = 123; // expected-note {{read of dereferenced null pointer}}
```

Comment:
This looks like a regression in the diagnostic, there's not a read happening there.

Reply:
I think we can either
  - say something like "access to dereferenced null pointer"
  - introduce a different diagnostic for this case, so we can just say "dereferencing null pointer"